app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/parser/OpeningHoursParser.kt

```diff
@@ -53,9 +53,30 @@ fun OpeningHoursRuleList.toOpeningHoursRows(): List<OpeningHoursRow>? {
     return result
 }
 
+/** returns null if the list of rules cannot be displayed by the opening hours widget */
+fun OpeningHoursRuleList.toCollectionTimesRows(): List<OpeningHoursRow>? {
+    if (!isSupportedCollectionTimes()) {
+        // parsable, but not supported by StreetComplete
+        return null
+    }
+
+    val result = mutableListOf<OpeningHoursRow>()
+
+    for (rule in rules) {
+        if (rule.modifier != null && rule.modifier!!.isSimpleOff()) {
+            result.add(rule.createOffDays())
+        } else {
+            result.addAll(rule.createOpeningWeekdays())
+        }
```

Comment:
1. I'd very much prefer if the opening hours model used in the view and collection times model used in the view is separate.
2. The collection times form does not support off days, does it?

Reply:
Ok, let's finish this. I've rebased the PR and made it consistent with recent changes in the opening hours quest. Off days are ignored now for collection times.

But I don't really know how to best use such a CollectionTimesRow data class here. I think it would still have to be a subclass of OpeningWeekdaysRow anyway to avoid a lot of duplication in the opening hours parser...